client/my-sites/plan-features-2023-grid/components/sticky-container.tsx

```diff
@@ -0,0 +1,84 @@
+import { Global, css } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useRef, type ElementType, useState, useLayoutEffect, ReactNode } from 'react';
+
+type Props = {
+	children: ( isStuck: boolean ) => ReactNode;
+	stickyClass?: string;
+	element?: ElementType;
+	stickyOffset?: number; // offset from the top of the scrolling container to control when the element should start sticking, default 0
+	topOffset?: number; // offset from the top of the scrolling container to control the position of the sticky element, default 0
+};
+
+const Container = styled.div< { topOffset: number } >`
+	position: sticky;
+	top: ${ ( props ) => props.topOffset + 'px' };
+	z-index: 1;
+`;
+
+export function StickyContainer( props: Props ) {
+	const { stickyOffset = 0, topOffset = 0, stickyClass = '', element = 'div', children } = props;
+
+	const stickyRef = useRef( null );
+	const [ isStuck, setIsStuck ] = useState( false );
+
+	/**
+	 * This effect sets the value of `isStuck` state when it detects that
+	 * the element is sticky.
+	 * The top property of the root margin is set at -1px (plus optional offset).
+	 * So when position:sticky takes effect, the intersection ratio will always be ~99%
+	 */
+	useLayoutEffect( () => {
+		const observer = new IntersectionObserver(
```

Comment:
TIL about [the IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). Very cool!